packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmSetInternal.kt

```diff
@@ -105,8 +110,11 @@ internal class ManagedRealmSet<E>(
                 if (isEmpty()) {
                     throw NoSuchElementException("Could not remove last element returned by the iterator: set is empty.")
                 }
-                val element = RealmInterop.realm_set_get(nativePointer, pos.toLong())
-                val erased = RealmInterop.realm_set_erase(nativePointer, element)
+
+                val erased = getterScope {
+                    val elem = RealmInterop.realm_set_get(nativePointer, pos.toLong(), allocRealmValueT())
+                    RealmInterop.realm_set_erase(nativePointer, elem)
+                }
```

Comment:
Yeah, and this can be done by embedding the erase within the set too 😅:

```
RealmInterop.realm_set_get(nativePointer, pos.toLong(), allocRealmValueT(), erase = True)
```

The RealmInterop layer is in our control, and we can define any function and any signature, it is not a 1-1 mirror with the C-API, but a helper layer to interact with it. As a platform-specific knows how to deal with each platform's particularities.

Still think trying to fix these few situations, where we need to reuse a realm_value_t, isn't enough to justify leaking the scopes... but we have already discussed this, don't make me talk! 🤪